doc/vhpidirect/examples/arrays.rst

```diff
@@ -0,0 +1,56 @@
+.. program:: ghdl
+.. _COSIM:VHPIDIRECT:Examples:arrays:
+
+Arrays
+######
+
+.. _COSIM:VHPIDIRECT:Examples:arrays:logicvectors:
+
+:cosimtree:`logicvectors <vhpidirect/arrays/logicvectors>`
+**************************************************************
+
+Commonly signals in VHDL are of a logic type or a vector thereof (``std_logic`` and ``std_logic_vector``), coming from the ``ieee.std_logic_1164.*`` package.
+These types can hold values other than high and low (``1`` and ``0``) and are enumerated as:
+
+0. 'U' 
+1. 'X' 
+2. '0' 
+3. '1' 
+4. 'Z' 
+5. 'W' 
+6. 'L' 
+7. 'H' 
+8. '-' 
+
+Because the number of enumeration values is less than 256, logic values are transported in 8 bit words (a ``char`` type in C) (as mentioned in :ref:`Restrictions_on_foreign_declarations`).
+Logic vectors, of a bounded size, are easily created in C and passed to VHDL as ``char[]``. VHDL receives pointers as ``access`` types, in this case an access of a subtype of std_logic_vector.
+
+In this example two declarations make handling logic values in C a bit easier:
+
+- Providing logic values in C as their enumeration numbers is simplified with the use of a matching enumeration, ``HDL_LOGIC_STATES``. 
+- Printing out a logic value's associated character is also simplified with the ``const char HDL_LOGIC_CHAR[]`` declaration.
+
+This example builds on the matrices example (:ref:`COSIM:VHPIDIRECT:Examples:quickstart:matrices:`). Foreign subprograms are declared that enable receiving the size of two different logic vectors as well as the vectors themselves from C. There is only one subprogram to get the size of both C arrays, and it takes in an integer to determine which array's size gets returned. 
```

Comment:
On the one hand, it's going to be `arrays/matrices`, instead of `quickstart:matrices`. On the other hand, `arrays/intvector` (#11, #12) feels like a better reference now, because it is focused on different procedures to use bounded 1D arrays.

In some sense, passing an array of `std_logic_vector` is similar to passing a multidimensional array of `std_logic`. However, the underlaying moral behind `arrays/matrices` can be: you don't need to manage `std_logic` from C, just pass integers or reals and convert them to fixed-point in VHDL.

Hence, we should find a use case where passing 2D arrays of `std_logic` is preferred over a 1D array of integers/doubles. Then, we would add either an additional example or a subexample to this one (or to matrices). Nonetheless, this is not prioritary for now. Given the amount of pending examples to be merged, let's focus on those first.

Reply:
I feel like this follows directly from the arrays/intvector for sure. Will update the ref.